src/components/RandomTopicCards/index.tsx

```diff
@@ -0,0 +1,61 @@
+import * as S from "../TopicCards/styled";
+import IconArrow from "@assets/icons/left-arrow.svg";
+import CardDeck from "../CardDeck";
+import useTopicCardNavigation from "../../hooks/useTopicCardNavigation";
+import { useRandomTopics } from "../../hooks/useRandomTopics";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+interface RandomTopicCardsProps {
+  onHasViewedAllCards?: (hasViewedAllCards: boolean) => void;
+}
+
+export default function RandomTopicCards({
+  onHasViewedAllCards,
+}: RandomTopicCardsProps) {
+  const navigate = useNavigate();
+  const { data, fetchNextPage } = useRandomTopics();
+  const topics = data?.pages.flatMap((page) => page.topics) || [];
+  const { currentIndex, visibleTopics, handleSwipe } = useTopicCardNavigation({
+    topics,
+    onHasViewedAllCards,
+  });
+
+  const handleExit = () => {
+    navigate("/");
+  };
+
+  // 현재 인덱스가 전체 토픽 개수의 60%에 도달하면 다음 페이지 로드
+  useEffect(() => {
+    if (currentIndex >= topics.length * 0.6) {
+      fetchNextPage();
+    }
+  }, [currentIndex, topics.length, fetchNextPage]);
```

Comment:
```suggestion
  useEffect(() => {
    if (currentIndex >= topics.length * 0.6 && !isFetching) {
      fetchNextPage();
    }
  }, [currentIndex, topics.length, fetchNextPage]);
```
같은 페이지로 네트워크 요청이 중복되는 것 같아서 `isFetching` 같은 속성 값 가져와서 중복 요청 방지해보면 좋을 것 같아여!

- 개선 전
![개선전](https://github.com/user-attachments/assets/14bf910b-44dc-4e3e-b31e-de52090472a8)

- 개선 후 
![개선후](https://github.com/user-attachments/assets/3c399758-814b-44c6-9cad-68e8294cd6c3)

Reply:
아 그러네요 조건이 부등호라.. 감사합니다